spock-specs/src/main/groovy/org/spockframework/specs/extension/SnapshotExtension.groovy

```diff
@@ -17,24 +21,63 @@ class SnapshotExtension implements IAnnotationDrivenExtension<Snapshot> {
   }
 
   @Override
-  void visitFieldAnnotation(Snapshot annotation, FieldInfo field) {
-    Assert.that(field.type.isAssignableFrom(Snapshotter), "Field must be of type Snapshotter")
-    field.parent.bottomSpec.addSetupInterceptor(new SnapshotInterceptor(annotation, field))
+  void visitSpecAnnotation(Snapshot annotation, SpecInfo specInfo) {
+    specInfo.features.each { visitFeatureAnnotation(annotation, it) }
+  }
+
+  @Override
+  void visitFeatureAnnotation(Snapshot annotation, FeatureInfo feature) {
+    def snapshotterFields = feature.spec.allFields.findAll {
```

Comment:
I find it a bit weird, that the feature annotation sets a field, is this intended to override the extension?

Reply:
I'm not sure what you mean by "to override the extension", but intentionally it was, yes.
If you don't like it, I can change it of course - given we keep the annotation on class and method level actually.

Reply:
I get that you want to re-use the annotation as inclusion filter, but on the other hand the resulting usage feels sub-optimal.

> I think if we want to support method level, then it should be a parameter annotation. Which also removes the need for this lookup to be here, and it can be moved to be once in `visitSpecAnnotation`.

Reply:
I went ahead and implemented the scanning in #1665 WDYT?